### ISSUE TITLE
Id Annotation isn't inherited by child classes

### ISSUE THREAD

**Issue by deusprogrammer** [dizitart/nitrite-database#37]
Perhaps an annotation similar to @MappedSuper in JPA so if I am using the same id field in each entity I can just extend a parent class instead of having to mark up an id field in every class.

Otherwise great library, much better than CouchbaseLite.  Especially since it doesn't require any native libraries.

**Comment by anidotnet** [comment]
Thanks for the kind words, I am glad you like it.

For the feature itself, I am working on it.

**Comment by anidotnet** [comment]
This feature should be available in 2.1.0.SNAPSHOT now. 

I have added a new annotation ```@InheritIndices``` that you should apply on the child class. Once applied it will consider all index related annotations ```@Id```, ```@Index```, ```@Indices``` through out the class hierarchy.

The feature works like as follows:

```java
@Index(value = "text", type = IndexType.Fulltext)
public class SuperDuperClass {
    public String text;
}

@Indices(
        @Index(value = "date", type = IndexType.Unique)
)
public class ParentClass extends SuperDuperClass {
    @Id
    public Long id;
    public Date date;
}

@InheritIndices
public class ChildClass extends ParentClass {
    public String name;
}

```
Now the ```ChildClass``` will have indices for fields - ```text```, ```date``` and ```id```.

**Comment by deusprogrammer** [comment]
Now that's dedication.  My hat's off to you sir.
